android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt

```diff
@@ -1169,6 +1170,59 @@ class JsonConverter {
             playerViewConfig = toPlayerViewConfig(json),
             pictureInPictureConfig = toPictureInPictureConfig(json.getMap("pictureInPictureConfig")),
         )
+
+        @JvmStatic
+        fun fromBufferLevel(bufferLevel: BufferLevel?): WritableMap? {
+            if (bufferLevel == null) {
+                return null
+            }
+
+            return Arguments.createMap().apply {
+                putDouble("level", bufferLevel.level)
+                putDouble("targetLevel", bufferLevel.targetLevel)
+                putInt(
+                    "media",
+                    when (bufferLevel.media) {
+                        MediaType.Audio -> 0
+                        MediaType.Video -> 1
+                    },
+                )
+                putInt(
+                    "type",
+                    when (bufferLevel.type) {
+                        BufferType.ForwardDuration -> 0
+                        BufferType.BackwardDuration -> 1
+                    },
+                )
+            }
+        }
+
+        @JvmStatic
+        fun fromRNBufferLevels(bufferLevels: RNBufferLevels?): WritableMap? {
+            if (bufferLevels == null) {
+                return null
+            }
+
+            return Arguments.createMap().apply {
+                putMap("audio", fromBufferLevel(bufferLevels.audio))
+                putMap("video", fromBufferLevel(bufferLevels.video))
+            }
+        }
+
+        /**
+         * Maps an integer value into the corresponding [BufferType] value.
+         *
+         * Currently `0` is mapped to [BufferType.ForwardDuration], and `1` to [BufferType.BackwardDuration].
+         * Other values fall back to [BufferType.ForwardDuration]
+         * @param value Integer value representing the [BufferType].
+         * @return The [BufferType] corresponding to [value].
+         */
```

Comment:
Most of this comment could be removed if instead of integer you use strings to represent the enums 🙂

Also I think you are missing `toMediaType` function.

Reply:
Good catch!
Address in 24f6205665202344e2ff3ec57cbdffb0b9f818b3

Reply:
I added a comment here: https://github.com/bitmovin/bitmovin-player-react-native/commit/24f6205665202344e2ff3ec57cbdffb0b9f818b3#r130703874

Reply:
Address in d2ff9eacb1334cb7ab04d2ddd020def9e522bd52

Reply:
Forgot to update docs: fix in 0ada2f8de55bda6c7a3c484b522c5f3a0eb4fac8